src/sentry/static/sentry/app/views/settings/organizationRelays/relays.tsx

```diff
@@ -10,69 +11,76 @@ import {Organization} from 'app/types';
 import ExternalLink from 'app/components/links/externalLink';
 import space from 'app/styles/space';
 import Button from 'app/components/button';
-import {IconDelete} from 'app/icons';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
 
-import Time from './time';
+import {Relay} from './types';
+import Content from './content';
+import {Add, Edit} from './dialogs';
 
 const RELAY_DOCS_LINK = 'https://getsentry.github.io/relay/';
 
-type Relay = {
-  public_key: string;
-  name: string;
-  created: string;
-  first_seen: string;
-  last_seen: string | null;
-  last_modified?: string;
-  description?: string;
-};
-
 type Props = {
   organization: Organization;
 } & RouteComponentProps<{orgId: string}, {}>;
 
 type State = AsyncComponent['state'] & {
   relays: Array<Relay>;
+  openAddDialog: boolean;
+  editRelay?: Relay;
 };
 
-const relaysMock: Array<Relay> = [
-  {
-    public_key: '1:bb:6e:af:66:b4:38:e0:62:83:62:15:22:7',
-    name: 'First key',
-    description: 'optional description for the key',
-    first_seen: '2020-02-07T15:17:00Z',
-    last_seen: '2020-02-07T15:17:00Z',
-    created: '2020-02-07T15:17:00Z',
-  },
-  {
-    public_key: '2:bb:6e:af:66:b4:38:e0:62:83:62:15:22:7',
-    name: 'Second key',
-    description: 'optional description for the key',
-    first_seen: '2020-02-07T15:17:00Z',
-    last_seen: '2020-02-07T15:17:00Z',
-    last_modified: '2020-02-07T15:17:00Z',
-    created: '2020-02-07T15:17:00Z',
-  },
-];
-
 class Relays extends AsyncComponent<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      relays: relaysMock,
-    };
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    return [['relays', `/organizations/${this.props.organization.slug}/relay-keys/`]];
   }
 
-  // TODO(Priscila): activate the code below as soon as the endpoint is provided
-  // getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
-  //   return [['relays', `/organizations/${this.props.organization.slug}/relay-keys`]];
-  // }
-
-  // handleDelete = (id: Relay['public_key']) => () => {};
-
-  handleAdd = () => {};
+  handleDelete = (publicKey: Relay['publicKey']) => async () => {
+    try {
+      await this.api.requestPromise(
+        `/organizations/${this.props.organization.slug}/relay-keys/${publicKey}/`,
+        {method: 'DELETE'}
+      );
+      addSuccessMessage('Successfully deleted relay public key');
+      this.setState(prevState => ({
+        relays: prevState.relays.filter(relay => relay.publicKey !== publicKey),
+      }));
+    } catch {
+      addErrorMessage('An unknown error occurred while deleting relay public key');
+    }
+  };
+
+  handleSave = async (data: Parameters<Add['props']['onSave']>[0]) => {
+    try {
```

Comment:
Do you need to indicate a loading state before sending the request?

Reply:
It's good to have..let me add it